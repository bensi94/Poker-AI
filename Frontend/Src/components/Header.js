// Dependencies
import React from 'react';
import { Route, Switch } from 'react-router';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Loading } from 'element-react';


// Project files
import FrontPage from './FrontPage';
import Login from './Login';
import Signup from './Signup';
import Navbar from './Navbar';
import Game from './Game';
import PrivateRoute from '../routers/PrivateRoute';
import PublicRoute from '../routers/PublicRoute';
import { checkUser } from '../actions/user';


export class Header extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        let page;
        if (this.props.isAuthenticated === undefined ||
            (this.props.isAuthenticated && !this.props.token)) {
            page = <Loading fullscreen/>;
            this.props.checkUser();
        } else {
            page = (
                <React.Fragment>
                    <Navbar />
                    <Switch>
                        <Route exact path="/" component={FrontPage} />
                        <PublicRoute exact path="/login" component={Login} />
                        <PublicRoute exact path="/signup" component={Signup} />
                        <PrivateRoute exact path="/table/:tableID" component={Game} />
                        <Route component={FrontPage} />
                    </Switch>
                </React.Fragment>
            );
        }
        return (
            { ...page }
        );
    }
}

Header.propTypes = {
    username: PropTypes.string,
    name: PropTypes.string,
    isAuthenticated: PropTypes.bool,
    token: PropTypes.string,
    error: PropTypes.shape({
        detail: PropTypes.string
    }),
    checkUser: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
    return {
        username: state.user.username,
        name: state.user.name,
        isAuthenticated: state.user.isAuthenticated,
        token: state.auth.token,
        error: state.user.error
    };
};

const mapDispatchToProps = (dispatch) => ({
    checkUser: () => dispatch(checkUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
