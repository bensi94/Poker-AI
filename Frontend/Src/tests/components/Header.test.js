import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../components/Header';

describe('Header Test suite', () => {
    it('Should render Header(snapshot) correctly', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Should have header Navbar', () => {
        const wrapper = shallow(<Header />);
        expect(wrapper.find('Navbar')).toHaveLength(1);
    });
});
