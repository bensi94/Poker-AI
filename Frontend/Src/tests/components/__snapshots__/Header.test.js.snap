// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test suite Should render Header(snapshot) correctly 1`] = `
<Fragment>
  <Navbar />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/signup"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Fragment>
`;
